hpx/parallel/algorithms/inner_product.hpp

```diff
@@ -290,22 +81,9 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
     inner_product(ExPolicy&& policy, InIter1 first1, InIter1 last1,
         InIter2 first2, T init)
     {
```

Comment:
This is a simple forwarding function now. The asserts are checked by the called `transform_reduce` algorithm.